### DISCUSSION TITLE
Pretty disappointed in this project - zero support

### DISCUSSION THREAD

**Discussion by jxs714** [chatwoot/chatwoot#8954] [Self hosted] [answered]
As the title says, pretty disappointed in this project. No clear installation instructions for docker. Spent over 1-2 weeks trying to figure out how to make it work, to finally figure out how to install it and emails do not work and the system half works by not saving settings properly. Reached out to support and no response in over 2 weeks. Not sure why you would make the software "open source" if you're not going to provide direction to the community when they have questions...

**Comment by sojan-official** [comment]
I'm really sorry to hear about the troubles you've been experiencing with getting Chatwoot set up and the lack of support response. We understand how frustrating this can be and appreciate your patience.

For support, we have several options listed on our [FAQ page](https://www.chatwoot.com/docs/self-hosted/faq/), which might offer immediate answers to common issues faced by our users. Regarding installation, our recommended method is using cwctl, detailed instructions for which can be found [here](https://www.chatwoot.com/docs/self-hosted/deployment/chatwoot-ctl). This method simplifies the process and is designed to help avoid the very issues you've encountered.

We acknowledge that other installation methods are provided for advanced users with a good understanding of Ruby on Rails applications. These options are there to offer flexibility but, indeed, might require a deeper dive to troubleshoot specific problems that arise.

The open-source community around Chatwoot is typically very supportive, with users helping each other through challenges. However, we recognize that certain installation issues can be unique and complex, making them difficult to resolve quickly. We strive to assist with these as promptly as possible, but some cases may take longer to diagnose and address.

To better assist you, `could you please provide the GitHub ticket or Discord query reference that hasn't been responded to?` This will enable our team to look into your issue directly and ensure it receives the attention it deserves.

Again, I apologize for the inconvenience and the experience you've had so far. We're committed to improving and hope to resolve your issues soon.

**Reply by jxs714** [reply]
Hi,

Thanks for your concern and for reaching back with an update. The only reason I do not install by cwctl is because I am using plesk and it will interfere with the already installed nginx, postgresl and redis. I am concerned to put the app as root into system and try to aim for dockers to protect the systems main core. Here was my previous ticket: https://github.com/orgs/chatwoot/discussions/8823 - I ended up working around it since i had no choice. As far as sidekiq, I ended up bypassing by manually approving users in admin without verification.  Please see the issues I am having below:

I installed the software and are having many weird issues such as the following:

**Docker container issue**
- Sidekiq not sending emails with admin recover password page. (I bypassed this by just approving the account). All messages stay stuck in sidekiq queue or failed

**In Chatwoot Dashboard**
- the "forward to email" email box that the software gives is "b8304e49946db699b05bf04b5f8daaa1@emailname@domain.com". I cannot forward my emails to this randomized email by chatwoot if the @ sign is before the emailname@domain.com. Not sure why the system decided to give this email to forward to. I am confused at this process

- I embedded the chat widget into another site and its giving me different errors in console as soon as I open the chat and type something:

POST https://subdomain.domain.com/api/v1/widget/conversations/toggle_typing?website_token=wPC664yWHhgyv1JwsVVGUp7v&cw_conversation=eyJhbGciOiJIUzI1NiJ9.eyJzb3VyY2VfaWQiOiJlYmMwMTQwZi0yYTQ2LTRjODgtYjgyOC01YWZmODJiZWEwMzkiLCJpbmJveF9pZCI6Mn0.ylY4SutziVumMkfiIS5V8Ddqiy5SJq8wIiDE5zhT2Xs 404 (Not Found)

POST https://subdomain.domain.com/api/v1/widget/messages?website_token=wPC664yWHhgyv1JwsVVGUp7v&cw_conversation=eyJhbGciOiJIUzI1NiJ9.eyJzb3VyY2VfaWQiOiJlYmMwMTQwZi0yYTQ2LTRjODgtYjgyOC01YWZmODJiZWEwMzkiLCJpbmJveF9pZCI6Mn0.ylY4SutziVumMkfiIS5V8Ddqiy5SJq8wIiDE5zhT2Xs&locale=en 500 (Internal Server Error)

POST https://subdomain.domain.com/api/v1/widget/conversations/toggle_typing?website_token=wPC664yWHhgyv1JwsVVGUp7v&cw_conversation=eyJhbGciOiJIUzI1NiJ9.eyJzb3VyY2VfaWQiOiJlYmMwMTQwZi0yYTQ2LTRjODgtYjgyOC01YWZmODJiZWEwMzkiLCJpbmJveF9pZCI6Mn0.ylY4SutziVumMkfiIS5V8Ddqiy5SJq8wIiDE5zhT2Xs 404 (Not Found)

Uncaught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received

I found out the reason being is because agent bots are enabled. I had to scavenge hunt all forums and support questions where users are only finding solutions on their own without developer support. Customers can reply or initiate a chat, but they are getting errors on chat and console triggering these errors. (screenshot attached). If i disable the bots, i do not get these issues. Also, another issue is the agent chat bubble is infinite (on the customer side where the customer thinks we are responding) no matter what settings I have. I could be away from computer and not in the chat window text box on admin side and it will still show that I am typing to customer.

![download](https://github.com/chatwoot/chatwoot/assets/4393073/a4501f8f-1902-4d31-8828-74c23162de16)

Here is my nginx directive

set $upstream 0.0.0.0:3000;

underscores_in_headers on;

location / {
	proxy_pass_header Authorization;
	proxy_pass http://$upstream$request_uri;
	proxy_set_header Upgrade $http_upgrade;
	proxy_set_header Connection "upgrade";
	proxy_set_header Host $host;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_set_header X-Forwarded-Ssl on; # Optional

	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

	proxy_http_version 1.1;
	proxy_set_header Connection "";

	proxy_buffering off;
	client_max_body_size 0;
	proxy_read_timeout 36000s;
	proxy_redirect off;
}

location /cable {
	proxy_pass_header Authorization;
	proxy_pass http://$upstream$request_uri;
	proxy_set_header Upgrade $http_upgrade;
	proxy_set_header Connection "upgrade";
	proxy_set_header Host $host;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_set_header X-Forwarded-Ssl on; # Optional

	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

	proxy_http_version 1.1;
	proxy_set_header Connection "";

	proxy_buffering off;
	client_max_body_size 0;
	proxy_read_timeout 36000s;
	proxy_redirect off;
}

location /api {
	proxy_pass_header Authorization;
	proxy_pass http://$upstream$request_uri;
	proxy_set_header Upgrade $http_upgrade;
	proxy_set_header Connection "upgrade";
	proxy_set_header Host $host;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_set_header X-Forwarded-Ssl on; # Optional

	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

	proxy_http_version 1.1;
	proxy_set_header Connection "";

	proxy_buffering off;
	client_max_body_size 0;
	proxy_read_timeout 36000s;
	proxy_redirect off;
}


I see in both github and discord that several others had this issue and asked how to fix it since more than 4 months ago and no solutions are being provided for most questions across all channels by developer support. I dont understand it. You say everyone has their own setup and you recommend cwtl, but this option is not ideal if you are running a webserver with several sites and softwares.

**Comment by sojan-official** [comment]
@jxs714 It looks like there are a few issues you've run into, and I appreciate you bringing them to our attention. Here's how we can tackle them together:

- For a smooth setup, we suggest sticking to the official installation guide. Our team knows it inside out, so we can offer quicker and more accurate support. If you opt for a different installation method, it might take a bit longer for the community to chime in. You can find a helpful installation video here: https://www.youtube.com/watch?v=srolHJskK5Q

- Chatwoot relies on various environment variables for full functionality, such as sending emails or uploading images. Check out our documentation to configure them correctly: https://www.chatwoot.com/docs/self-hosted/configuration/environment-variables/

- As for the response bot, it's currently in beta and isn't enabled by default. If you've encountered an infinite typing indicator, it's likely due to incomplete configuration settings for this feature. I'd recommend disabling it until you're able to complete the setup.

**Reply by jxs714** [reply, marked as answer]
Hi @sojan-official 

Thank you for reaching back. I had to test a bit more with confidence and patience, and I figured out how to get Chatwoot fully working. Sorry, configuring Chatwoot has been a frustrating ride but I finally got it to work with the setup. Here is my setup

Software
Plesk (latest)
Postgresql (Locally, not in docker)
Redis (Locally, not in docker)
Rails : Setup as Port xxx1 (whichever port you want) - Docker container
Sidekiq: Setup as Port xxx2 (whichever port you want) - Docker container
Base: (no need, this can be deleted and is just a template for rails to build off of) - Docker container

For the docker IP to connect to redis and postgresql locally (or any local software), the IP needs to be 172.17.0.1 and port is the same for the services (ie: redis 6379 and postgresql 5432)

Overall, I didn't realize I needed both the docker for Rails and Sidekiq running at once, but on different ports. This way Sidekiq connects to Rails / Chatwoot and runs properly for admin and user emails outside of the main admin dashboard. This also fixes issues with chat not working and giving errors in the browser. For anyone needing a custom docker-compose,yaml file, here is a configuration that should fix your setup issues if youre using plesk. I only set up Rails and Sidekiq (along with the temporary "base" container)

`version: '3'

services:
  base: &base
    image: chatwoot/chatwoot:latest
    env_file: .env ## Change this file for customized env variables
    volumes:
      - /data/storage:/app/storage

  rails:
    <<: *base
    ports:
       - '127.0.0.1:4000:3000' #The 4000 port is what I chose, you can change this
    environment:
      - NODE_ENV=production
      - RAILS_ENV=production
      - INSTALLATION_ENV=docker
    entrypoint: docker/entrypoints/rails.sh
    command: ['bundle', 'exec', 'rails', 's', '-p', '3000', '-b', '0.0.0.0']

  sidekiq:
    <<: *base
    ports:
       - '127.0.0.1:4001:3000' #The 4001 port is what I chose, you can change this
    environment:
      - NODE_ENV=production
      - RAILS_ENV=production
      - INSTALLATION_ENV=docker
    command: ['bundle', 'exec', 'sidekiq', '-C', 'config/sidekiq.yml']
`

Thank you guys and keep up the great work
